File: typescript/9_Palindrome_Number.ts
function isPalindrome(x: number): boolean {
  let arr = []
  let digit = String(x).length
  console.log(`digit: ${digit}`)
  let quotient = 0 // 商
  let remainder = x // 余り

  for (let i = digit - 1; i >= 0; i--) {
    quotient = Math.floor(remainder / 10 ** i)
    remainder = x % 10 ** i
    console.log(`quotient: ${quotient}`)
    console.log(`remainder: ${remainder}`)
    arr.push(quotient)
    console.log(`debug: arr: ${arr}`)
  }

  console.log(arr)
  const org = arr.slice()
  const rev = arr.reverse()
  console.log(org)
  console.log(rev)
  console.log(org.toString())
  console.log(rev.toString())

  if (org.toString() === rev.toString()) {
    return true
  } else {
    return false
  }
}

// true
if (isPalindrome(121)) {
  console.log('ok')
}

// false
if (!isPalindrome(-121)) {
  console.log('ok')
}

// false
if (!isPalindrome(10)) {
  console.log('ok')
}

File: ruby/9_Palindrome_Number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
  return false if x < 0

  num = x
  str = num.to_s
  rev = str.reverse
  puts str
  puts rev
  if str == rev
    return true
  else
    return false
  end
end

puts 'ok' if is_palindrome(121)
puts 'ok' unless is_palindrome(-121)
puts 'ok' unless is_palindrome(10)
